src/Katla.idr

```diff
@@ -9,6 +9,45 @@ import Libraries.Data.PosMap
 import Data.List1
 import Data.List
 import Data.String
+import Data.SnocList
+
+laTeXHeader : String
+laTeXHeader =  """
+\newcommand{\IdrisHlightFont}         {\ttfamily}
+\newcommand{\IdrisHlightStyleData}    {}
+\newcommand{\IdrisHlightStyleType}    {}
+\newcommand{\IdrisHlightStyleBound}   {}
+\newcommand{\IdrisHlightStyleFunction}{}
+\newcommand{\IdrisHlightStyleKeyword} {\bfseries}
+\newcommand{\IdrisHlightStyleImplicit}{\itshape}
+\newcommand{\IdrisHlightStyleComment} {\itshape}
+\newcommand{\IdrisHlightStyleHole}    {\bfseries}
+
+\newcommand{\IdrisHlightColourData}    {IndianRed1}
+\newcommand{\IdrisHlightColourType}    {DeepSkyBlue3}
+\newcommand{\IdrisHlightColourBound}   {DarkOrchid3}
+\newcommand{\IdrisHlightColourFunction}{Chartreuse4}
+\newcommand{\IdrisHlightColourKeyword} {black}
+\newcommand{\IdrisHlightColourImplicit}{Darkorchid3}
+
+\newcommand{\IdrisHlightColourComment} {grey}
+\newcommand{\IdrisHlightColourHole}    {yellow}
+
+\newcommand{\IdrisHole}[1]{{%
+    \colorbox{yellow}{%
+      \IdrisHlightStyleHole\IdrisHlightFont%
+      #1}}}
+
+\newcommand{\RawIdrisHighlight}[3]{{\textcolor{#1}{#2\IdrisHlightFont#3}}}
```

Comment:
The highlighting is not designed to go over linebreaks. The annotations are for each smallest term in a source. Aside from String Literals none of these would span multiple lines.

Reply:
Comments (which we currently don't yet treat in the semantic highlighter, but we might in the future) would also span multiple lines.

Reply:
So the issue is that `fGetLine` returns the newline characters.
If an annotation doesn't span multiple lines, this is fine, because the newline character(s) at the end of the line would have `Nothing` as decoration, and so the `Nil` case of `processLine` will output them as plain characters and end the line.

But if an annotation does span multiple lines, the first line gets printed without annotation in the `Nil` case of `processLine`.

The messed up highlighting seems to be a bug in the semantic annotation tool.

Reply:
(will fix the PR accordingly, thanks.)